hwy/ops/scalar-inl.h

```diff
@@ -1784,6 +1784,21 @@ HWY_API Vec1<int32_t> WidenMulPairwiseAdd(D32 /* tag */, Vec1<int16_t> a,
   return Vec1<int32_t>(a.raw * b.raw);
 }
 
+// ------------------------------ SatWidenMulPairwiseAdd
+
+#ifdef HWY_NATIVE_U8_I8_SATWIDENMULPAIRWISEADD
+#undef HWY_NATIVE_U8_I8_SATWIDENMULPAIRWISEADD
+#else
+#define HWY_NATIVE_U8_I8_SATWIDENMULPAIRWISEADD
+#endif
+
+template <class DI16, HWY_IF_I16_D(DI16)>
+HWY_API Vec1<int16_t> SatWidenMulPairwiseAdd(DI16 /* tag */, Vec1<uint8_t> a,
+                                             Vec1<int8_t> b) {
+  return Vec1<int16_t>(static_cast<int16_t>(a.raw) *
```

Comment:
I think we'd have to add saturation here before passing to the Vec1 ctor. Or perhaps just remove this specialization and let generic_ops handle it?

Reply:
If `a` is a 1-lane uint8_t vector and `b` is a 1-lane int8_t vector, `a[0] * b[0]` is within the range of an int16_t as `a[0] * b[0]` is between -32640 and 32385.

`a[0] * b[0] + a[1] * b[1]` can only overflow if `a[0]`, `a[1]`, `b[0]`, and `b[1]` are all non-zero and `b[0]` and `b[1]` both have the same sign.

Reply:
Oh, I see; it is only the addition that can overflow, and it doesn't happen in the scalar target. Thanks for explaining and adding the helpful comment 👍